src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -421,7 +448,73 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     }
   }
 
-  override def handleCallRestarts(restartableWorkflows: Seq[RestartableWorkflow])(implicit ec: ExecutionContext): Future[Any] = Future("FIXME")
+  /**
+   * <ul>
+   *   <li>Any execution in Failed should fail the restart.</li>
+   *   <li>Any execution in Aborted should fail the restart.</li>
+   *   <li>Scatters in Starting should fail the restart.</li>
+   *   <li>Collectors in Running should be set back to NotStarted.</li>
+   *   <li>Calls in Starting should be rolled back to NotStarted.</li>
+   *   <li>Calls in Running with no job key should be rolled back to NotStarted.</li>
+   * </ul>
+   *
+   * Calls in Running *with* a job key should be left in Running.  The WorkflowActor is responsible for
+   * resuming the call actors for these calls.
+   */
+  override def prepareForRestart(restartableWorkflow: WorkflowDescriptor)(implicit ec: ExecutionContext): Future[Unit] = {
+
+    lazy val tag = s"Workflow ${restartableWorkflow.id.shortString}:"
+
+    def handleExecutionStatuses(executions: Traversable[Execution]): Future[Unit] = {
+
+      val executionsByStatus = executions.groupBy(_.status) map { case (k, v) => ExecutionStatus.withName(k) -> v }
+      def stringifyExecutions(executions: Traversable[Execution]): String = {
+        executions.toSeq.sortWith((lt, rt) => lt.callFqn < rt.callFqn || (lt.callFqn == rt.callFqn && lt.index < rt.index)).mkString(" ")
+      }
+
+      if (executionsByStatus.contains(ExecutionStatus.Failed)) {
+        val failedExecutions = stringifyExecutions(executionsByStatus.get(ExecutionStatus.Failed).get)
+        Future.failed(new Throwable(s"$tag Cannot restart, found executions in Failed status: " + failedExecutions))
+      }
+      else if (executionsByStatus.contains(ExecutionStatus.Aborted)) {
+        val abortedExecutions = stringifyExecutions(executionsByStatus.get(ExecutionStatus.Aborted).get)
+        Future.failed(new Throwable(s"$tag Cannot restart, found executions in Aborted status: " + abortedExecutions))
+      }
+      else {
+        // Cromwell currently does not persist the types of executions.  Scatters are identified by this magic
+        // "$scatter_" string.
+        val (scatters, nonScatters) = executions partition { _.callFqn.contains("$scatter_") }
```

Comment:
Not sure if this is the same thing you're talking about, but I do see `$scatter`s in the database:

```
mysql> select * from EXECUTION where CALL_FQN like '%$scatter%' LIMIT 3;
+--------------+-----------------------+--------------------+--------+------+------+----------+--------+
| EXECUTION_ID | WORKFLOW_EXECUTION_ID | CALL_FQN           | STATUS | IDX  | RC   | START_DT | END_DT |
+--------------+-----------------------+--------------------+--------+------+------+----------+--------+
|            2 |                     1 | sc_test.$scatter_0 | Done   |   -1 |    0 | NULL     | NULL   |
|           10 |                     2 | sc_test.$scatter_0 | Done   |   -1 |    0 | NULL     | NULL   |
|           18 |                     3 | sc_test.$scatter_0 | Done   |   -1 |    0 | NULL     | NULL   |
+--------------+-----------------------+--------------------+--------+------+------+----------+--------+
3 rows in set (0.00 sec)
```